boards/arm/stm32f103_mini/doc/index.rst

```diff
@@ -0,0 +1,162 @@
+
++-----------+------------+-------------------------------------+
+| Interface | Controller | Driver/Component                    |
++===========+============+=====================================+
+| NVIC      | on-chip    | nested vector interrupt controller  |
++-----------+------------+-------------------------------------+
+| UART      | on-chip    | serial port-polling;                |
+|           |            | serial port-interrupt               |
++-----------+------------+-------------------------------------+
+| PINMUX    | on-chip    | pinmux                              |
++-----------+------------+-------------------------------------+
+| GPIO      | on-chip    | gpio                                |
++-----------+------------+-------------------------------------+
+| CLOCK     | on-chip    | reset and clock control             |
++-----------+------------+-------------------------------------+
+| FLASH     | on-chip    | flash memory                        |
++-----------+------------+-------------------------------------+
+| WATCHDOG  | on-chip    | independent watchdog                |
++-----------+------------+-------------------------------------+
+| ADC       | on-chip    | ADC Controller                      |
++-----------+------------+-------------------------------------+
+
+Other hardware features are not yet supported in this Zephyr port.
+
+The default configuration can be found in the defconfig file:
+``boards/arm/stm32f103_mini/stm32f103_mini_defconfig``
+
+Connections and IOs
+===================
+
+Each of the GPIO pins can be configured by software as output (push-pull or open-drain), as
+input (with or without pull-up or pull-down), or as peripheral alternate function. Most of the
+GPIO pins are shared with digital or analog alternate functions. All GPIOs are high current
+capable except for analog inputs.
+
+Board connectors:
+-----------------
+.. image:: img/stm32f103_mini_pin.jpg
+   :width: 800px
+   :align: center
+   :height: 619px
+   :alt: Nucleo F103RB connectors
+
+Default Zephyr Peripheral Mapping:
+----------------------------------
+
+- UART_1 TX/RX : PA9/PA10
+- UART_2 TX/RX : PA2/PA3 (ST-Link Virtual COM Port)
+- SPI1 NSS/SCK/MISO/MOSI : PA4/PA5/PA6/PA7 (Arduino SPI)
+- SPI2 SCK/MISO/MOSI : PB12/PB13/PB14/PB15
+- I2C1 SDA/SCL: PB9/PB8 (Arduino I2C)
+- PWM1_CH1: PA8
+- USER_PB : PC13
+- LD1 : PA5
+
+Programming and Debugging
+*************************
+
+Applications for the ``stm32f103_mini`` board configuration can be built and
+flashed in the usual way (see :ref:`build_an_application` and
+:ref:`application_run` for more details).
+
+Flashing
+========
+
+There are 2 main entry points for flashing STM32F1X SoCs, one using the ROM
+bootloader, and another by using the SWD debug port (which requires additional
+hardware). Flashing using the ROM bootloader requires a special activation
+pattern, which can be triggered by using the BOOT0 pin.
```

Comment:
In (boards/arm/stm32f103_mini/board.cmake)[https://github.com/zephyrproject-rtos/zephyr/pull/30177/commits/13dceada50ac2df204c90a96dba881332acca677#diff-f4cb4e26b5274d7bc7b78b8a6d0d651c330ccfe0ef4585422f9f79cba7736376], you configured openocd/jlink, which is differet from what you're describing here. Can you clarify how this is supposed to work?

Reply:
> In (boards/arm/stm32f103_mini/board.cmake)[https://github.com/[/pull/30177](https://github.com/zephyrproject-rtos/zephyr/pull/30177)/commits/13dceada50ac2df204c90a96dba881332acca677#diff-f4cb4e26b5274d7bc7b78b8a6d0d651c330ccfe0ef4585422f9f79cba7736376], you configured openocd/jlink, which is differet from what you're describing here. Can you clarify how this is supposed to work?

Hi, in **board.cmake**, I configured openocd / st-link which uses the SWD debug port as described in **index.rts**.

And the other method uses serial port (RX/TX) which requires tools from ST as described here:

[http://manpages.ubuntu.com/manpages/cosmic/man1/st-flash.1.html](http://manpages.ubuntu.com/manpages/cosmic/man1/st-flash.1.html)

Is it OK to include only st-link in board which is a much better choice, because st-flash does not support Debug?

Reply:
Sure, ST-link is fine, but please mention it in index.rst